microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from pathlib import PurePath
+from typing import Any, List, Optional, Type
+
+from lisa.executable import Tool
+from lisa.operating_system import CBLMariner
+from lisa.tools import Docker, Echo, Git, Modprobe
+
+
+class CloudHypervisorTests(Tool):
+    TIME_OUT = 1800
+
+    repo = "https://github.com/cloud-hypervisor/cloud-hypervisor.git"
+
+    cmd_path: PurePath
+    repo_root: PurePath
+
+    @property
+    def command(self) -> str:
+        return str(self.cmd_path)
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Git, Docker]
+
+    def run_tests(self, test_type: str, skip: Optional[List[str]] = None) -> List[str]:
+        self.node.tools[Modprobe].load("openvswitch")
+
+        if skip is not None:
+            skip_args = " ".join(map(lambda t: f"--skip {t}", skip))
+        else:
+            skip_args = ""
+
+        result = self.run(
+            f"tests --{test_type} -- -- {skip_args}",
+            timeout=self.TIME_OUT,
+            force_run=True,
+            cwd=self.repo_root,
```

Comment:
shell=True,
            sudo=True,

Reply:
It has been working fine during my testing. I know for sure that sudo is not required to run these tests.

Reply:
Have you triggered this testing? I encounter the permission issue here.

Reply:
here is the error
```
2022-06-16 16:27:41.274[8984][DEBUG] lisa.env[generated_0].node[0].cmd[1102] cmd: ['/home/l****t/lisa_working/tool/cloudhypervisortests/cloud-hypervisor/scripts/dev_cli.sh', 'tests', '--integration', '--', '--'], cwd: /home/l****t/lisa_working/tool/cloudhypervisortests/cloud-hypervisor, shell: False, sudo: False, posix: True, remote: True
2022-06-16 16:27:42.274[7348][INFO] lisa.env[generated_0].node[0].cmd[1102].stdout Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.24/images/create?fromImage=cloudhypervisor%2Fdev&tag=20220524-0: dial unix /var/run/docker.sock: connect: permission denied
2022-06-16 16:27:42.384[7348][INFO] lisa.env[generated_0].node[0].cmd[1102].stdout Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.24/build?buildargs=%7B%22TARGETARCH%22%3A%22amd64%22%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&rm=1&shmsize=0&t=cloudhypervisor%2Fdev%3A20220524-0&target=dev&ulimits=null&version=1: dial unix /var/run/docker.sock: connect: permission denied        
2022-06-16 16:27:42.384[7348][INFO] lisa.env[generated_0].node[0].cmd[1102].stdout [Cloud Hypervisor] Error pulling/building container image. Aborting.
```

Reply:
This happens on Ubuntu but not on Mariner. But using sudo messes up the permissions of all the files in the cloud-hypervisor directory and causes problems in Mariner. The correct way to solve this is to add the current user to the `docker` group. I will implement this.

Reply:
Fixed.